.pipelines/e2e.yaml

```diff
@@ -3,78 +3,19 @@ trigger: none
 pr:
   branches:
     include:
-    - master
-    - official/*
+      - master
+      - official/*
   paths:
     include:
-    - .pipelines/e2e.yaml
-    - e2e
-    - parts/linux
-    - pkg/agent
+      - .pipelines/e2e.yaml
+      - .pipelines/templates/e2e-template.yaml
+      - e2e
+      - parts/linux
+      - pkg/agent
     exclude:
-    - e2e/windows
-    - pkg/agent/datamodel/sig_config*.go # SIG config changes
-    - pkg/agent/datamodel/*.json # SIG version changes
-    - pkg/agent/testdata/AKSWindows* # Windows test data
-    
-variables:
-- group: ab-e2e
-- name: timeout
-  value: 90
-- name: parallel
-  value: 100
-
-pool:
-  name: nodesigtest-pool
-
+      - e2e/windows
+      - pkg/agent/datamodel/sig_config*.go # SIG config changes
+      - pkg/agent/datamodel/*.json # SIG version changes
+      - pkg/agent/testdata/AKSWindows* # Windows test data
 jobs:
-- job: Run_AgentBaker_E2E
-  timeoutInMinutes: ${{ variables.timeout }}
-  steps:
-  - bash: |
-      az login --identity
-      az account set -s $(AZURE_SUBSCRIPTION_ID)
-    displayName: az login
-  - bash: bash .pipelines/scripts/setup_go.sh
-    displayName: setup go
-  - bash: |
-      export PATH="/usr/local/go/bin:$PATH"
-      go version
-      
-      echo "VHD_BUILD_ID=$VHD_BUILD_ID"
-
-      if [ -n "${VHD_BUILD_ID}" ]; then
-        export SIG_VERSION_TAG_NAME=buildId
-        export SIG_VERSION_TAG_VALUE=$VHD_BUILD_ID
-        export IGNORE_SCENARIOS_WITH_MISSING_VHD=true
-    
-        # You can add other commands here that you want to execute
-      else
-        echo "Build.BuildId is not specified. Running default e2e tests."
-      fi
-
-      cd e2e
-      go test -parallel ${{ variables.parallel}} -timeout ${{ variables.timeout }}m -v -run Test_All ./
-    displayName: Run AgentBaker E2E
-    env:
-      VHD_BUILD_ID: $(VHD_BUILD_ID)
-      LOCATION: $(LOCATION)
-  - publish: $(System.DefaultWorkingDirectory)/e2e/scenario-logs
-    artifact: scenario-logs
-    condition: always()
-  - bash: |
-      set -x
-
-      vmssResourceIds=""
-      for vmssModel in e2e/scenario-logs/*/vmssId.txt; do
-        resourceId=$(cat ${vmssModel})
-        vmssResourceIds="${vmssResourceIds} ${resourceId}"
-      done
-
-      if [ -n "${vmssResourceIds// }" ]; then
-        az resource delete --ids ${vmssResourceIds}
-      fi
-    displayName: ensure vmss deletion
-    condition: always()
- 
- 
+  - template: ./templates/e2e-template.yaml
```

Comment:
can we add a new boolean parameter as a "feature flag" to determine whether we run the e2e's?

Reply:
Where would you use it?

e2e tests are run after the build. So, a failure there doesn't change much.

Reply:
shouldnt the e2e prevent an official release from happening ?   Are we saying that the e2e failures will not present the release of the build artifact ? (for now ?) 

@cameronmeissner  what is your concern ? I'm guessing before the e2e were not executed as part of the build ? I thought this was just ensuring that the e2e would be using the latest images that was just built ? unless you are worried about the fact that e2e now run during PR pipeline ?

Reply:
I think the main thing is that we don't currently run abe2e's in AME (and might require some extra work to do so), so having a flag we can easily use to turn on/off the e2e's might be useful

Reply:
My current goal is to run e2e on each PR. With a goal to reduce manual testing and to avoid merging broken things.

I think release is a more manual process I don't know much about, so I can't really suggest the e2e automation there.

Reply:
@cameronmeissner Do you have an example of a specific pipeline run where executing the e2e test would cause any issues?

I don't fully understand the issue, so I can't come up with a good solution at the moment. It’s quite easy to disable in the future. I suggest delaying the attempt to solve potential issues until we encounter them.